aws_health_to_slack/program.py

```diff
@@ -0,0 +1,115 @@
+"""Announce AWS health events in Slack, based on resource ownership in Google Sheet.
+
+Documentation:
+https://docs.aws.amazon.com/health/
+https://aws.amazon.com/blogs/mt/tag/aws-health-api/
+"""
+
+from datetime import UTC, datetime, timedelta
+import json
+import os
+import re
+
+import autokitteh
+from autokitteh.aws import boto3_client
+from autokitteh.google import google_sheets_client
+from autokitteh.slack import slack_client
+
+
+url = os.getenv("GOOGLE_SHEET_URL")
+
+
+def on_every_minute(_):
+    """Workflow's entry-point."""
+    slack_channels = _read_google_sheet()
+    events = _aws_health_events()
+    events_by_arn = {event["arn"]: event for event in events}
+
+    for entity in _affected_aws_entities(events):
+        project = entity.get("tags", {}).get("project")
+        if not project:
+            print(f"Error: AWS entity without project tag: {entity}")
+            continue
+
+        channel = slack_channels.get(project)
+        affecting_events = [events_by_arn[arn] for arn in entity["eventArns"]]
+        _post_slack_message(project, channel, entity, affecting_events)
+
+
+@autokitteh.activity
+def _read_google_sheet() -> dict[str, str]:
+    """Read mapping of project tags to Slack channels from Google Sheet."""
+    match = re.match(r"(.*/d/)?([\w-]{20,})", url)
```

Comment:
I don't have a specific idea in mind, but thinking about the general user and whether or not they're going to understand regular expressions. It might be intimidating.

Reply:
I think it's a great idea!

Reply:
See https://github.com/autokitteh/autokitteh/pull/458